### PR TITLE
doc(rfc): Add requirements / non requirements section to RFC.

### PR DESCRIPTION
**Pull request by jplaisted**
This seems helpful to reviewers (and probably also the author). Motivation is too high level to be able to properly review a design. We need a clear set of requirements to review the design against.

Stems from comments on https://github.com/linkedin/datahub/pull/1778#pullrequestreview-461847439

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: docs/rfc/templates/000-template.md [hunk 1]
```diff
@@ -21,6 +21,28 @@
...
+## Requirements
+
+> What specific requirements does your design need to meet? This should ideally be a bulleted list of items you wish
+> to achieve with your design. This can help everyone involved (including yourself!) make sure your design is robust
+> enough to meet these requirements.
+>
+> Once everyone has agreed upon the set of requirements for your design, we can use this list to review the detailed
+> design.
+
+### Extensibility
```
Comment (keremsahin1) [inline docs/rfc/templates/000-template.md:33]: Why is this a sub-section of requirements? Shouldn't it be top level or a sub-section of detailed design?

File: docs/rfc/templates/000-template.md [hunk 2]
```diff
@@ -21,6 +21,28 @@
 > alternative solutions. In other words, enumerate the constraints you are trying to solve without coupling them too
 > closely to the solution you have in mind.
 
+## Requirements
```
Comment (keremsahin1) [inline docs/rfc/templates/000-template.md:24]: Do you think `Goals` & `Non-goals` would be better?

File: docs/rfc/templates/000-template.md (hunk 1 above)
Comment (jplaisted) [reply docs/rfc/templates/000-template.md:33]: Extensibility _requirements_? I'm fine either way, but this a little clearer is it still a list of requirements.

File: docs/rfc/templates/000-template.md (hunk 2 above)
Comment (jplaisted) [reply docs/rfc/templates/000-template.md:24]: No, those can be too high level imo. "Goals" says high level, "requirements" says specific, to me at least.

For most designs I really think we need a detailed, bulleted list. We need to consider the design's requirements/goals in detail from the outset, and in plain English (outside of detailed design).

**Comment by jplaisted** [comment]
> Let's also add a deadline section? Depending on the size of RFC, we can have 1 week for small, 2 weeks for medium, and 4 weeks for large. The reviewers can request extensions (maybe 1 week at a time?) if the discussion isn't converging. This way we can move RFCs forward at a more predictable pace.

Should that be in the RFC or in the PR/issue? Either way probably involves updating the process doc. And can be done in a different PR :)

File: docs/rfc/templates/000-template.md (hunk 1 above)
Comment (hshahoss) [reply docs/rfc/templates/000-template.md:33]: Would be good to have this as a separate section as part of "Future Work".

**Comment by mars-lan** [comment]
> > Let's also add a deadline section? Depending on the size of RFC, we can have 1 week for small, 2 weeks for medium, and 4 weeks for large. The reviewers can request extensions (maybe 1 week at a time?) if the discussion isn't converging. This way we can move RFCs forward at a more predictable pace.
> 
> Should that be in the RFC or in the PR/issue? Either way probably involves updating the process doc. And can be done in a different PR :)

Ideally both, but definitely for RFC. PR should really just be a straight forward realization of RFC and should be less controversial when it comes to review time.

**Comment by jplaisted** [comment]
> > > Let's also add a deadline section? Depending on the size of RFC, we can have 1 week for small, 2 weeks for medium, and 4 weeks for large. The reviewers can request extensions (maybe 1 week at a time?) if the discussion isn't converging. This way we can move RFCs forward at a more predictable pace.
> > 
> > 
> > Should that be in the RFC or in the PR/issue? Either way probably involves updating the process doc. And can be done in a different PR :)
> 
> Ideally both, but definitely for RFC. PR should really just be a straight forward realization of RFC and should be less controversial when it comes to review time.

I think my concern is discoverability. I don't want an RFC opened with a week deadline and then just an immediate submit a week later. So if you aren't getting comments, I'm not sure putting it in the review is the best place to put it? I think we need more of a process around it. So I think we can do this in a different PR.
